cmd/geth/consolecmd.go

```diff
@@ -218,12 +216,9 @@ func ephemeralConsole(ctx *cli.Context) error {
 			utils.Fatalf("Failed to execute %s: %v", file, err)
 		}
 	}
-	// Wait for pending callbacks, but stop for Ctrl-C.
-	abort := make(chan os.Signal, 1)
-	signal.Notify(abort, syscall.SIGINT, syscall.SIGTERM)
 
 	go func() {
-		<-abort
+		stack.Wait()
```

Comment:
Actually, I think the fix would be even better if, `os.Exit(0)` were replaced by `console.Stop(false)`

Reply:
done